docs/use-in-web-environment.md

```diff
@@ -0,0 +1,408 @@
+```
+
+The code above will result in:
+
+```html
+<head>
+    <script>var username = "John";</script>
+</head>
+<body>
+    <script>var age = 42;</script>
+</body>
+```
+
+The values of the variables will be encoded, using the `Yiisoft\Json\Json::htmlEncode()` method:
+
+```php
+$view->registerJs('data', ['username' => 'John', 'age' => 42]);
+```
+
+The code above will result in:
+
+```html
+<script>var data = {"username":"John","age":42};</script>
+```
+
+The name of variable will be used as key, preventing duplicated variable names.
+
+## Use with an asset manager
+
+If you are using the [yiisoft/assets](https://github.com/yiisoft/assets) package, the `Yiisoft\View\WebView`
+class provides methods for batch adding CSS and JavaScript data.
+
+```php
+/**
+ * @var Yiisoft\View\WebView $this
+ * @var Yiisoft\Assets\AssetManager $assetManager
+ */
+ 
+$assetManager->register([
+    MyAsset::class,
+]);
+
+$this->addCssFiles($assetManager->getCssFiles());
+$this->addCssStrings($assetManager->getCssStrings());
+$this->addJsFiles($assetManager->getJsFiles());
+$this->addJsStrings($assetManager->getJsStrings());
+$this->addJsVars($assetManager->getJsVars());
+```
+
+These methods process the CSS and JavaScript configuration created by the asset manager and converts it into HTML code.
+
+## WebView Events
+
+The `Yiisoft\View\WebView` class trigger several events during the view rendering process. Events are classes:
+
+- `Yiisoft\View\Event\WebView\BeforeRender` - triggered at the beginning of rendering a view template.
+  Listeners of this event may set `$event->stopPropagation()` to be false to cancel the rendering process.
+- `Yiisoft\View\Event\WebView\AfterRender` - triggered at the ending of rendering a view template.
+  Listeners of this event can get data about the rendering results using the methods of this class.
+- `Yiisoft\View\Event\WebView\PageBegin` - triggered by the call of `Yiisoft\View\WebView::beginPage()` in the view template.
+- `Yiisoft\View\Event\WebView\PageEnd` - triggered by the call of `Yiisoft\View\WebView::endPage()` in the view template.
+- `Yiisoft\View\Event\WebView\Head` - triggered by the call of `Yiisoft\View\WebView::head()` in the view template.
+- `Yiisoft\View\Event\WebView\BodyBegin` - triggered by the call of `Yiisoft\View\WebView::beginBody()` in the view template.
+- `Yiisoft\View\Event\WebView\BodyEnd` - triggered by the call of `Yiisoft\View\WebView::endBody()` in the view template.
+
+These events are passed to the `Psr\EventDispatcher\EventDispatcherInterface` implementation,
+which is specified in the constructor when the `Yiisoft\View\WebView` instance is initialized.
+
+## Security
+
+When creating views that generate HTML pages, it is important that you encode and/or filter the data
+coming from end users before presenting them. Otherwise, your application may be subject to
+[cross-site scripting](http://en.wikipedia.org/wiki/Cross-site_scripting) attacks.
```

Comment:
```suggestion
When creating views that generate HTML pages, it is important that you properly encode and/or filter all the data when outputted. Otherwise, your application may be subject to
[cross-site scripting](http://en.wikipedia.org/wiki/Cross-site_scripting) attacks.
```